marlowe-dashboard-client/src/MainFrame/View.purs

```diff
@@ -17,12 +17,14 @@ render :: forall m. MonadAff m => State -> ComponentHTML Action ChildSlots m
 render state =
   let
     currentSlot = view _currentSlot state
-
-    toast = view _toast state
   in
     div [ classNames [ "h-full" ] ]
-      [ case view _subState state of
-          Left pickupState -> PickupAction <$> renderPickupState pickupState
-          Right playState -> PlayAction <$> renderPlayState currentSlot playState
-      , ToastAction <$> renderToast toast
+      -- Before adding the tooltips (https://github.com/input-output-hk/plutus/pull/3440)
+      -- this view was a case statement, selecting one subview, or the other. Now, we use
+      -- the fact that _pickupState and _playState are mutually exclusive so we
+      -- "render both" at the same times. Because of how renderSubmodule works, if
+      -- the lens is not pointing at anything, an empty div will be drawn instead.
```

Comment:
What is the advantage of this approach? I am asking this because the need of a comment explaining something that was explicit syntactically before seems to me like it would be a downgrade

Reply:
Well, before we had 
`Right playState -> PlayAction <$> renderPlayState currentSlot playState` which works for an `HTML a b` but not with a `ComponentHTML a b c`. The reason is that for a simple `HTML` we can just map the action wrapper, but for a `ComponentHTML` we need to do a `bimap (map actionWrapper) actionWrapper`, which is what `renderSubmodule` does.

My first approach in refactoring this was:
```purescript
case view _subState state of
          Left pickupState -> renderSubmodule _pickupState PickupAction renderPickupState state
          Right playState -> renderSubmodule _playState PlayAction (renderPlayState currentSlot) state
      , ToastAction <$> renderToast toast
```
but then, because we are using `renderSubmodule` we have no use for pattern matching the state.

Reply:
well, as a nice side effect of the current version, if we change from Either to SubState this code wouldn't change as long as we update the _playState and _pickupState lens :).

And even if I use the case, I can't use the inner state as renderSubmodule expects the whole state and zooms into the lens, so it would be 

```purescript
case view _subState state of
          Left _ -> renderSubmodule _pickupState PickupAction renderPickupState state
          Right _ -> renderSubmodule _playState PlayAction (renderPlayState currentSlot) state
      , ToastAction <$> renderToast toast
```

vs 

```purescript
renderSubmodule _pickupState PickupAction renderPickupState state
renderSubmodule _playState PlayAction (renderPlayState currentSlot) state
renderSubmodule _toast ToastAction renderToast state
```

I don't mind either way